hw/ip/edn/doc/programmers_guide.md

```diff
@@ -19,6 +19,47 @@ void edn_init(unsigned int enable) {
 
 EDN may only be enabled if CSRNG is enabled.
 Once disabled, EDN may only be re-enabled after CSRNG has been disabled and re-enabled.
+The only exception to this is when firmware takes care of properly uninstantiating the associated CSRNG instance before disabling EDN.
+EDN can then be safely re-enabled without disabling and re-enabling CSRNG first.
+For details, refer to [Uninstantiating CSRNG instance through EDN](#uninstantiating-csrng-through-edn).
+
+The recommended enable sequence for the entropy complex is to first enable ENTROPY_SRC, then CSRNG, and finally EDN.
+
+## Uninstantiating CSRNG through EDN
+
+To disable and re-enable EDN without disabling and re-enabling CSRNG, firmware must ensure the associated CSRNG instance is properly uninstantiated.
+Otherwise, EDN and the associated CSRNG application interface may get out of sync at which point both EDN and CSRNG have to be disabled and re-enabled.
+This means also unrelated CSRNG instances and EDNs get disabled and then need to be re-configured.
```

Comment:
Grammar nit: I think you probably want "This means unrelated CSRNG instances and EDNs also get disabled..."